src/esp/assets/GenericInstanceMeshData.cpp

```diff
@@ -44,11 +46,20 @@ struct InstancePlyData {
    * semantic mesh for better frustum culling.
    */
   bool objIdsFromPly = false;
+  bool objPartitionsFromSSD = false;
+
+  /**
+   * @brief This is used to map every vertex to a particular partition, for
+   * culling.
+   */
+  std::vector<uint16_t> partitionIds;
 };
 
 Cr::Containers::Optional<InstancePlyData> parsePly(
```

Comment:
> GenericInstanceMeshData looks to be a pretty simple container for mesh data. Would it make sense to keep this class simple and move all your annotation-parsing logic to a separate class and file?

This class was always used for the semantic mesh processing - the annotation code that I added just adds flexibility for different ply formats than just the hard-coded mp3d support.

For long-term handling considerations, i see this class holding only specialized annotation information (if that - i can see a future without it too) while the general mesh data is handled by GenericMeshData, if not Mn::Trade::MeshData directly.

> Related: it would help if this class GenericInstanceMeshData had a class-level comment. Its parent class says "Base class for storing mesh asset data including geometry and topology" and I so wouldn't expect such a class to know e.g. the details of region indices in annotation data.

Again, I just expanded existing annotation functionality in this class to handle the specific needs of the Hm3D annotations - previously it used hardcoded vert ids in the ply mesh to denote submeshes for culling.  I can certainly more clearly comment the purpose of this class, but like I said, from an organizational standpoint, it would probably be better if this class only held semantic-specific functionality, expanding on existing whatever MeshData structure ended up being the backbone for this class.

A little bit of redesign/refactor would pay big dividends in this department, but I felt such engineering overhead was beyond the scope of this PR.

Reply:
>I felt such engineering overhead was beyond the scope of this PR

I will push back a little here on the idea "refactors are out of scope". You're adding nearly 1000 lines of code in this PR. I believe we should view such large changes as opportunities to refactor, rather than piling on more tech debt to an area of code that we already regard as needing refactor.

>it would probably be better if this class only held semantic-specific functionality
Consider renaming the class in this case.

Reply:
> I will push back a little here on the idea "refactors are out of scope". You're adding nearly 1000 lines of code in this PR. I believe we should view such large changes as opportunities to refactor, rather than piling on more tech debt to an area of code that we already regard as needing refactor.

I completely share your concerns, and normally would be in total agreement regarding addressing refactors and tech-debt whenever possible. 

However, the support for the Appen annotated hm3d meshes was presented to me as being urgent, and so I approached implementation as the priority. Refactors were restricted to what was necessary to support the new functionality in as generalized a way as possible while not breaking existing code (theoretically) and addressing some of the brittleness inherent in the old designs.  There's very little tech debt that's been added - the new functionality is required for hm3d annotation support, so any future reworks are going to need it on some level;  some (very small amount) of tech debt has been removed, and, unfortunately a lot of debt remains.

This affected how multiple classes performed their duties:
 - Renderer (instance shaders on construction instead of the previous method of create shader on first visualize call, expand colormap support)
 - GenericMeshLoading (support multiple vert color formats, support no baked-in ids, better multi-mesh culling support)
 - Sim/ResourceManager (create SemanticScene descriptor in resource manager to make available during Semantic Scene mesh load) 
 
 
> Consider renaming the class in this case

Good idea.